src/dmd/expression.d

```diff
@@ -6431,9 +6439,11 @@ extern (C++) final class CondExp : BinExp
         return new CondExp(loc, econd.syntaxCopy(), e1.syntaxCopy(), e2.syntaxCopy());
     }
 
-    override int checkModifiable(Scope* sc, int flag)
+    override Modifiable checkModifiable(Scope* sc, int flag)
     {
-        return e1.checkModifiable(sc, flag) && e2.checkModifiable(sc, flag);
+        if (e1.checkModifiable(sc, flag) && e2.checkModifiable(sc, flag))
```

Comment:
Note that checkModifiable() returns a 3-state enum, not a boolean. Please rewrite as checking against the enum state.